docs/querying/functions.md

```diff
@@ -432,6 +432,23 @@ by the number of seconds under the specified time range window, and should be
 used primarily for human readability.  Use `rate` in recording rules so that
 increases are tracked consistently on a per-second basis.
 
+## `info()`
+
+**This function has to be enabled via the [feature flag](../feature_flags.md#experimental-promql-functions) `--enable-feature=promql-experimental-functions`.**
+
+For each time series in `v`, `info(v instant-vector, [data-label-selector string])` finds 
+all `target_info` series with corresponding identifying labels (`instance` and `job`),
+and adds the union of their data (i.e., non-identifying) labels to the time series.
+`target_info`'s data labels correspond to OpenTelemetry resource attributes.
+The optional second argument allows specifying through label matchers, which data labels
+should be picked. Example: `{k8s_container_name=~".+"}`. Through this argument, you can
+also specify the info series to consider, by providing a `__name__` label matcher, e.g.
+`{__name__="target_info"}`. For the time being however, only `target_info` will be considered.
+The set of time series set augmented with any matching info series labels gets returned.
+
+In the future, `info` will recognize all info metrics, not just `target_info`, and also detect their 
+respective identifying labels.
```

Comment:
I think we need to make clearer what we try to accomplish with this experiment. And we cannot really assume that the reader has any idea what an identifying label and a data label is.

Maybe we start with how the function is supposed to work, and then describe the restriction in the current MVP.

I'll try to come up with a concrete suggestion tomorrow…

Reply:
As discussed with @beorn7 in person, I can try to come up with a better documentation myself.

Reply:
Yeah, "tomorrow" was overly optimistic… It's still on my list to propose a text, but whoever gets here first should let the others know about their idea.

Reply:
Starting on it now, last week flew by.

Reply:
@beorn7 I rewrote the documentation, what do you think of the current revision, does it match what you have in mind? Do you want me to make any edits?

Reply:
Still working through my backlog. I hope I'll get to this soon.